lib/layout/ConnectionDocking.js

```diff
@@ -33,29 +33,27 @@ export default function ConnectionDocking() {}
 /**
  * Return the actual waypoints of the connection (visually).
  *
- * @param {djs.model.Connection} connection
- * @param {djs.model.Base} [source]
- * @param {djs.model.Base} [target]
+ * @param {djs.model.Connection|Array<Point>} connectionOrWaypoints
+ * @param {djs.model.Shape} [source]
+ * @param {djs.model.Shape} [target]
  *
  * @return {Array<Point>}
  */
-ConnectionDocking.prototype.getCroppedWaypoints = function(connection, source, target) {
-  return connection.waypoints;
+ConnectionDocking.prototype.getCroppedWaypoints = function(connectionOrWaypoints, source, target) {
+  return connectionOrWaypoints.waypoints || connectionOrWaypoints;
 };
 
 /**
  * Return the connection docking point on the specified shape
  *
- * @param {djs.model.Connection} connection
+ * @param {Array<Point>} waypoints
  * @param {djs.model.Shape} shape
  * @param {Boolean} [dockStart=false]
  *
  * @return {DockingPointDescriptor}
  */
-ConnectionDocking.prototype.getDockingPoint = function(connection, shape, dockStart) {
-
-  var waypoints = connection.waypoints,
-      dockingIdx,
+ConnectionDocking.prototype.getDockingPoint = function(waypoints, shape, dockStart) {
```

Comment:
Breaking change?

Reply:
An alternative would be to accept both a connection and waypoints. The reason I changes this is that I want to be able to get the docking point from a list of waypoints so I don't have to monkey-patch them in dmn-js while I'm doing this.

Reply:
Are we keeping track of the breaking changes? As long as we do it, we should be fine.

I do not see a `BREAKING CHANGE` comment in any of your commits though.

Reply:
I've added the comment.

Reply:
I've removed this commit. See https://github.com/bpmn-io/diagram-js/pull/453#discussion_r402909819

Reply:
Thanks.